app/gui/view/component-browser/component-group/src/new_entry.rs

```diff
@@ -0,0 +1,454 @@
+/// selection extending a bit over the background what looks bad.
+const CONTOUR_PADDING: f32 = 0.5;
+
+
+
+// ==================
+// === Background ===
+// ==================
+
+/// The background of the Component Browser Entry. It consists of a rectangle and an optional shadow
+/// underneath it. The shadow is shown under the group headers which are pushed down to be visible
+/// in the viewport.
+pub mod background {
+    use super::*;
+
+    // We don't use the usual padding of sprites, because we clip the shadow under the background
+    // and clipping the shadow shape with `*` operator causes glitches.
+    // See https://www.pivotaltracker.com/story/show/182593513
+
+    ensogl::define_shape_system! {
+        below = [grid_view::entry::overlay, grid_view::selectable::highlight::shape];
+        pointer_events = false;
+        (style:Style, color:Vector4, height: f32, shadow_height_multiplier: f32) {
+            let color = Var::<color::Rgba>::from(color);
+            let width: Var<Pixels> = "input_size.x".into();
+            let height: Var<Pixels> = height.into();
+            let bg = Rect((&width, &height)).fill(color);
+            // We use wider and shorter rect for the shadow because of the visual artifacts that
+            // will appear otherwise:
+            // 1. Rounded corners of the shadow are visible if the rect is too narrow. By widening
+            //    it we keep the shadow sharp and flat for the whole width of the header.
+            // 2. Visual glitching similar to z-fighting occurs on the border of the elements
+            //    when the shadow rect has the exact same size as the background. We shrink the
+            //    height by 1 pixel to avoid it.
+            let shadow_rect = Rect((width * 2.0, height - 1.0.px()));
+            let mut shadow_parameters = shadow::parameters_from_style_path(style, theme::header::shadow);
+            shadow_parameters.size = shadow_parameters.size * shadow_height_multiplier;
+            let shadow = shadow::from_shape_with_parameters(shadow_rect.into(), shadow_parameters);
+            (shadow + bg).into()
+        }
+    }
+}
+
+
+
+// =============
+// === Model ===
+// =============
+
+// === Kind ===
+
+/// The kind of entry:
+#[derive(Clone, Copy, Debug, Default, Eq, PartialEq)]
+pub enum Kind {
+    /// A standard entry with icon and caption.
+    #[default]
+    Entry,
+    /// A group header; there is no icon, the text id stronger and there is a gap above header
+    /// being a group separator.
+    Header,
+    /// The entry in LocalScope section which leave no gap to the left or right.
+    LocalScopeEntry,
+}
+
+
+// === Model ===
+
+/// The [model](grid_view::entry::Entry) for Component Browser Entry.
+#[allow(missing_docs)]
+#[derive(Clone, Debug, Default)]
+pub struct Model {
+    pub kind:        Kind,
+    pub color:       color::Rgba,
+    pub caption:     ImString,
+    pub highlighted: Rc<Vec<text::Range<text::Bytes>>>,
```

Comment:
Does it make sense to indicate the range in bytes? Or should this be something higher level, like `Character`/`Glyph`? I seem to remember the new text implementation to support such indexing, but I’m not entirely sure.

Reply:
The text::Area API with non-byte indexing is not published yet, so I would stay with this and fix perhaps in the [integration task](https://www.pivotaltracker.com/story/show/182610422)